bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx

```diff
@@ -196,6 +196,21 @@ static bool IsCTypesArrayOrPointer(PyObject* pyobject)
 }
 
 
+//- helper to establish life lines -------------------------------------------
+static inline bool SetLifeLine(PyObject* holder, PyObject* target, intptr_t ref)
+{
+// set a lifeline from on the holder to the target, using the ref as label
+    if (!holder) return false;
+
+// 'ref' is expected to be the converter address or data memory location, so
+// that the combination of holder and ref is unique, but also identifiable for
+// reuse when the C++ side is being overwritten
+    std::ostringstream attr_name;
+    attr_name << "__" << ref;
+    auto res = PyObject_SetAttrString(holder, (char*)attr_name.str().c_str(), target);
+    return res != -1;
```

Comment:
Reading at the the signature of [PyObject_SetAttrString](https://docs.python.org/3/c-api/object.html#c.PyObject_SetAttrString), I have the following two questions:
1. The second argument accepts a `const char *`, which is already the return type of `attr_name.str().c_str()`, is the extra `(char*)` needed?
2. The return value of `PyObject_SetAttrString` should be `0` on success and `1` on failure according to the documentation. If this is true and there are no other cases not reported in the docs, these lines could be substituted by
```suggestion
    return PyObject_SetAttrString(holder, attr_name.str().c_str(), target) == 0
```

Reply:
Ok for 1, and I'll break the `str().c_str()` chain. 2 is fine either way :smile:

Reply:
Ad 1.: the `(char*)` is for older pythons and still appears in older code throughout CPyCppyy, so is retained for consistency. Ad 2.: it's `-1` on failure, not `1` and it is likewise by convention: `-1` is the common failure report across the C-API, whereas `0` is _not_ a common success report across same.